### DISCUSSION TITLE
Shape Optimization in MACH-Aero framework/ADFLOW/pyOptSparse

### DISCUSSION THREAD

**Discussion by bhargavisrinivas666** [pyg-team/pytorch_geometric#9363] [Q&A]
I have done aerodynamic shape optimization of a wing successfully, But I 'm struggling to do wing optimization by including body/fuselage. I want to optimize the wing by keeping the body/fuselage. I generated surface mesh of my geometry in pointwise and volume mesh in pyHYP by using overset approach. I used the same flow and optimizations conditions as I used for wing alone case.I added FFD points of body FFD box as fixed.by using addLocationConstraint().
I'm getting this below error. Once the baseline geometry aerodynamic analysis is done.

Run-time error in procedure metric
Error message: Negative volumes present in grid.

What might be the reason for this error,As there are no negative volumes in my input mesh?
 Can anyone help me with this?
Thanks in advance.

**Comment by rusty1s** [comment]
I am afraid I am of not much help here. Is this a PyG issue?

**Reply by bhargavisrinivas666** [reply]
I'm not sure about the reason for this issue.

And while giving the Volume Constraint,we have to mention leList and teList.Should I keep the wing constraints only as I'm focusing on wing optimization or I have to include the body part also?If yes how can I include there is only option as leList and teList?

How can I fix the control points?.As I don't want any geometrical changes to be made for the body .
